source/agora/consensus/protocol/Nominator.d

```diff
@@ -347,17 +353,24 @@ extern(D):
 
     /***************************************************************************
 
-        Begins the nomination timer
+        Arm the nomination timer
 
         This should be called once when a Node's enrollment has been confirmed
-        on the blockchain.
+        on the blockchain or after checkNominate task completes.
 
     ***************************************************************************/
 
-    public void startNominatingTimer () @trusted nothrow
+    public void startNominatingTimer () @safe nothrow
     {
         log.info("Starting nominating timer..");
-        this.timers[TimersIdx.Nomination].rearm(this.nomination_interval, false);
+        startTaskTimer(TimersIdx.Nomination, this.nomination_interval);
+    }
+
+    private void startTaskTimer (in TimersIdx timer_id, in Duration interval) @trusted nothrow
```

Comment:
Should have a parameter for `periodic` having `false`  as defaults.

Reply:
As in `Nominator` we always rearm the timer after the `Task` and do not use the periodic option I would prefer to not include it in this private function.

Reply:
I see what you mean.